### PR TITLE
Update configuration to work with new actuator

### PR DESCRIPTION
**Pull request by bison**
This is just a quick fix so we can start testing with a newer actuator image. We need to spend more time thinking about the types here and exposing more of the available options.

### CODE REVIEW

File: pkg/render/machine-api-operator-config.yaml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 apiVersion: v1
```
Comment (enxebre) [inline pkg/render/machine-api-operator-config.yaml]: We should move this into the examples folder

File: manifests/clusterapi-controller.yaml [hunk 2]
```diff
@@ -45,14 +45,14 @@ spec:
             cpu: 100m
             memory: 30Mi
       - name: aws-machine-controller
-        image: quay.io/alberto_lamela/aws-machine-controller:mvp # TODO: move this to openshift org
+        image: quay.io/bison/aws-machine-controller:cfc14e5 # TODO: move this to openshift org
```
Comment (enxebre) [inline manifests/clusterapi-controller.yaml]: We should be able to use image built by CI now? cc @ingvagabund

File: machines/machine-set.yaml [hunk 3]
```diff
@@ -1,40 +1,53 @@
...
-      versions:
-        kubelet: 0.0.0
-        controlPlane: 0.0.0
-      roles:
-      - Master
-
+          ami:
+            id: "{{ .MachineConfig.AMI }}"
+          credentialsSecret:
+            name: aws-credentials-secret
```
Comment (enxebre) [inline machines/machine-set.yaml]: This is not available on the installer, we should omit it so the aws client uses the machine ami role

File: manifests/clusterapi-controller.yaml (hunk 2 above)
Comment (ingvagabund) [reply manifests/clusterapi-controller.yaml]: Can't find the image yet.

**Comment by enxebre** [comment]
See follow up for consolidating config here https://jira.coreos.com/browse/CLOUD-138
In general I think we should expose as config only which is strictly necessary for integrating with the installer.
Did you have the chance to adapt to this and test it via https://github.com/openshift/installer/pull/119?

File: machines/machine-set.yaml (hunk 3 above)
Comment (bison) [reply machines/machine-set.yaml]: Makes sense. This is nice for testing locally with the AWS actuator though. It should probably just be optional.

**Comment by bison** [comment]
I didn't get a chance to test via the installer yet. I'll give that a shot.

File: manifests/clusterapi-controller.yaml (hunk 2 above)
Comment (bison) [reply manifests/clusterapi-controller.yaml]: I see `machine-api-operator` images in the output of `oc describe is -n openshift origin-v3.11 --server https://api.ci.openshift.org` but they reference the internal registry on that cluster. The images are supposed to be mirrored to Quay and Docker Hub, but I'm not sure where to find them.

File: machines/machine-set.yaml (hunk 3 above)
Comment (bison) [reply machines/machine-set.yaml]: Pushed a quick fix to make this optional.

File: pkg/render/machine-api-operator-config.yaml (hunk 1 above)
Comment (bison) [reply pkg/render/machine-api-operator-config.yaml]: Done.

**Comment by enxebre** [comment]
closed via https://github.com/openshift/machine-api-operator/pull/35

**Pull request closed without merging**